src/utils/utils.go

```diff
@@ -186,6 +186,26 @@ func GetHostIpsFromInventory(inventory *models.Inventory) ([]string, error) {
 	return ips, nil
 }
 
+func WaitForPredicateWithTimer(ctx context.Context, timeout time.Duration, interval time.Duration, predicate func(timer *time.Timer) bool) error {
+	timeoutTimer := time.NewTimer(timeout)
```

Comment:
defer timeoutTimer.Stop()

Reply:
maybe we should defer ticket.stop too? i think i missed it too

Reply:
can we change usage of wait with context :
`func WaitForPredicateWithContext(ctx context.Context, timeout time.Duration, interval time.Duration, predicate func() bool) error {
	return WaitForPredicateWithTimer(ctx, timeout, interval, func(timer *time.Timer) bool {
		return predicate()
	})
}

func WaitForPredicateWithTimer(ctx context.Context, timeout time.Duration, interval time.Duration, predicate func(timer *time.Timer) bool) error {
	timeoutTimer := time.NewTimer(timeout)
	ticker := time.NewTicker(interval)
	// Keep trying until we're time out or get true
	for {
		select {
		case <-ctx.Done():
			return ctx.Err()
		// Got a timeout! fail with a timeout error
		case <-timeoutTimer.C:
			return errors.New("timed out")
		// Got a tick, we should check on checkSomething()
		case <-ticker.C:
			if predicate(timeoutTimer) {
				return nil
			}
		}
	}
}`